staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -34,6 +34,16 @@ type CustomResourceDefinitionSpec struct {
 	Validation *CustomResourceValidation
 	// Subresources describes the subresources for CustomResources
 	Subresources *CustomResourceSubresources
+
+	// Columns
+	Columns []Column
+}
+
+type Column struct {
+	// Header is the header to be printed on-top of the column.
+	Header string
+	// Path is a simple JSON path, i.e. with array notation.
+	Path string
```

Comment:
Do we want to support `Type string` with OpenAPI types? Currently every cell is returned as string.

Reply:
I'd start with strings, for now. We can always improve if needed.

Reply:
We can't actually return new cells of different types in the future unless we define it now because that would not be backwards compatible for a client that assumes strings.  We need to up front force people to acknowledge non-strings if we ever want to support them.